### PR TITLE
Build on mac

### PR DESCRIPTION
**Pull request by robinbb**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,3 +27,16 @@ RUN sed -i 's/nixbld:x:30000:.*/nixbld:x:30000:/g' /etc/group
 ADD entrypoint.py /
 ENTRYPOINT ["/entrypoint.py"]
 CMD ["/bin/bash"]
+
+# LeapYear changes below this point, only:
```
Comment (facundominguez) [inline Dockerfile]: These changes need more justification in comments. Why should the docker image contain a build of sparkle?

File: stack.yaml [hunk 2]
```diff
@@ -1,17 +1,23 @@
+resolver: lts-7.8
+
 packages:
 - .
+- location: inline-java/jni
+  extra-dep: true
+- location: inline-java/jvm
+  extra-dep: true
```
Comment (facundominguez) [inline stack.yaml]: An inline-java folder seems to be missing here.

File: shell.nix [hunk 3]
```diff
@@ -7,10 +7,10 @@
           sha256 = "04530jl2chy0klima5mby3pqq2xwgnm3mhv7gxzc4wa5s6i42q6h";
           };
      in import pkgs {}
-, ghc ? nixpkgs.haskell.compiler.ghc7103   # Default needed for Docker build.
+, ghc ? nixpkgs.ghc
 }:
 
-with nixpkgs;
+with nixpkgs // { ghc = ghc; };
```
Comment (facundominguez) [inline shell.nix]: Unnecessary?
```
$ nix-instantiate --eval --expr 'let x = 2; in let x = 3; in with {x = 1;}; x'
3
```
Also, `with nixpkgs;` doesn't shadow `ghc` for me.

File: stack.yaml (hunk 2 above)
Comment (facundominguez) [inline stack.yaml]: ok, but there be a submodule to provide the contents of those locations.

File: Dockerfile (hunk 1 above)
Comment (robinbb) [inline Dockerfile]: This Dockerfile is not merely one to use to build Sparkle. It is a build of Sparkle.

File: shell.nix (hunk 3 above)
Comment (robinbb) [inline shell.nix]: You're right. With-expressions in Nix are funny things. I will remove.

File: stack.yaml (hunk 2 above)
Comment (robinbb) [inline stack.yaml]: @facundominguez This PR includes that submodule.

**Comment by robinbb** [comment]
@facundominguez Please have another look to see if we can merge this PR.

**Comment by facundominguez** [comment]
The Dockerfile changes need some discussion still. But the rest of the patch could be merged right away.

**Comment by robinbb** [comment]
@facundominguez Okay, the PR requires that you approve it. If it can be merged, please approve and merge.

**Pull request merged**